src/mlpack/tests/ne_test.cpp

```diff
@@ -0,0 +1,238 @@
+
+  // Test set function.
+  linkGene.Weight(20);
+  BOOST_REQUIRE_EQUAL(linkGene.Weight(), 20);
+
+  // Test copy constructor.
+  LinkGene linkGene2(linkGene);
+  BOOST_REQUIRE_EQUAL(linkGene2.FromNeuronId(), 1);
+
+  // Test operator =.
+  LinkGene linkGene3(1, 3, 1, 7, true);
+  linkGene = linkGene3;
+  BOOST_REQUIRE_EQUAL(linkGene3.InnovationId(), 1);
+
+}
+
+/**
+ * Test NeuronGene class.
+ */
+BOOST_AUTO_TEST_CASE(NENeuronGeneTest)
+{
+  // Create a neuron gene.
+  NeuronGene neuronGene(1, INPUT, SIGMOID, 0, std::vector<double>(), 0, 0);
+
+  // Test parametric constructor and access functions.
+  BOOST_REQUIRE_EQUAL(neuronGene.Id(), 1);
+  BOOST_REQUIRE_EQUAL(neuronGene.Type(), INPUT);
+  BOOST_REQUIRE_EQUAL(neuronGene.ActFuncType(), SIGMOID);
+  BOOST_REQUIRE_EQUAL(neuronGene.Input(), 0);
+  BOOST_REQUIRE_EQUAL(neuronGene.Activation(), 0);
+
+  // Test set function
+  neuronGene.ActFuncType(RELU);
+  BOOST_REQUIRE_EQUAL(neuronGene.ActFuncType(), RELU);
+
+  // Test copy function.
+  NeuronGene neuronGene2(neuronGene);
+  BOOST_REQUIRE_EQUAL(neuronGene2.Id(), 1);
+
+  // Test operator =.
+  NeuronGene neuronGene3(11, INPUT, SIGMOID, 0, std::vector<double>(), 0, 0);
+  neuronGene = neuronGene3;
+  BOOST_REQUIRE_EQUAL(neuronGene.Id(), 11);
+
+}
+
+/**
+ * Test Genome class.
+ */
+BOOST_AUTO_TEST_CASE(NEGenomeTest)
+{
+  mlpack::math::RandomSeed(1);
+
+  // Construct seed genome for xor task.
+  int id = 0;
+  int numInput = 3;
+  int numOutput = 1;
+  double fitness = -1;
+  std::vector<NeuronGene> neuronGenes;
+  std::vector<LinkGene> linkGenes;
+
+  NeuronGene inputGene1(0, INPUT, LINEAR, 0, std::vector<double>(), 0, 0);
+  NeuronGene inputGene2(1, INPUT, LINEAR, 0, std::vector<double>(), 0, 0);
+  NeuronGene biasGene(2, BIAS, LINEAR, 0, std::vector<double>(), 0, 0);
+  NeuronGene outputGene(3, OUTPUT, SIGMOID, 1, std::vector<double>(), 0, 0);
+  NeuronGene hiddenGene(4, HIDDEN, SIGMOID, 0.5, std::vector<double>(), 0, 0);
+
+  neuronGenes.push_back(inputGene1);
+  neuronGenes.push_back(inputGene2);
+  neuronGenes.push_back(biasGene);
+  neuronGenes.push_back(outputGene);
+  neuronGenes.push_back(hiddenGene);
+
+  LinkGene link1(0, 3, 0, 1, true);
+  LinkGene link2(1, 3, 0, 1, true);
+  LinkGene link3(2, 3, 0, 0.5, true);
+  LinkGene link4(0, 4, 0, -0.5, true);
+  LinkGene link5(1, 4, 0, 2, true);
+  LinkGene link6(2, 4, 0, 1, true);
+  LinkGene link7(4, 3, 0, 0.1, true);
+
+  linkGenes.push_back(link1);
+  linkGenes.push_back(link2);
+  linkGenes.push_back(link3);
+  linkGenes.push_back(link4);
+  linkGenes.push_back(link5);
+  linkGenes.push_back(link6);
+  linkGenes.push_back(link7);
+
+  Genome seedGenome = Genome(0, 
+                             neuronGenes,
+                             linkGenes,
+                             numInput,
+                             numOutput,
+                             fitness);
+
+  // Test seed genome.
+  std::vector<std::vector<double>> inputs;  // TODO: use arma::mat for input.
```

Comment:
It would be great if we could to that change before we merge the code in. If you like I can do that for you.